sources/packages/api/src/esdc-integration/e-cert-full-time-integration/e-cert-full-time-response.service.ts

```diff
@@ -0,0 +1,144 @@
+import { Injectable, LoggerService } from "@nestjs/common";
+import { InjectLogger } from "../../common";
+import {
+  DisbursementScheduleErrorsService,
+  DisbursementScheduleService,
+} from "../../services";
+import { ProcessSftpResponseResult } from "./models/e-cert-full-time-integration.model";
+import { ECertFullTimeIntegrationService } from "./e-cert-full-time-integration.service";
+import { ECertResponseRecord } from "./e-cert-files/e-cert-response-record";
+import { getUTCNow } from "../../utilities";
+
+@Injectable()
+export class ECertFullTimeResponseService {
+  constructor(
+    private readonly disbursementScheduleService: DisbursementScheduleService,
+    private readonly eCertFullTimeService: ECertFullTimeIntegrationService,
+    private readonly disbursementScheduleErrorsService: DisbursementScheduleErrorsService,
+  ) {}
+  /**
+   * Download all files from E-Cert Response folder on SFTP and process them all.
+   * @returns Summary with what was processed and the list of all errors, if any.
+   */
+  async processResponses(): Promise<ProcessSftpResponseResult[]> {
+    const filePaths =
+      await this.eCertFullTimeService.getResponseFilesFullPath();
+    const processFiles: ProcessSftpResponseResult[] = [];
+    for (const filePath of filePaths) {
+      processFiles.push(await this.processFile(filePath));
+    }
+    return processFiles;
+  }
+
+  /**
+   * Process each individual E-Cert response file from the SFTP.
+   * @param filePath E-Cert response file to be processed.
+   * @returns Process summary and errors summary.
+   */
+  private async processFile(
+    filePath: string,
+  ): Promise<ProcessSftpResponseResult> {
+    const result = new ProcessSftpResponseResult();
+    result.processSummary.push(`Processing file ${filePath}.`);
+
+    let responseFile: ECertResponseRecord[];
+
+    try {
+      responseFile = await this.eCertFullTimeService.downloadResponseFile(
+        filePath,
+      );
+    } catch (error) {
+      this.logger.error(error);
+      result.errorsSummary.push(`Error downloading file ${filePath}. ${error}`);
+      // Abort the process nicely not throwing an exception and
+      // allowing other response files to be processed.
+      return result;
+    }
+
+    result.processSummary.push(`File contains ${responseFile.length} records.`);
+
+    for (const feedbackRecord of responseFile) {
+      try {
+        await this.processErrorCodeRecords(feedbackRecord);
+        result.processSummary.push(
+          `Status record from line ${feedbackRecord.lineNumber}.`,
```

Comment:
I am not able to understand the value of this log.
Does it make sense to kee it?

Reply:
updated as discussed